### ISSUE TITLE
Improved security for Broadcast backups

### ISSUE THREAD

**Issue by flocke** [andOTP/andOTP#157]
In the latest release the BroadcastReceiver for the plain text backups has been disabled due to the fact that every 3rd-party app would be able to just dump the users OTP accounts in plain text.

In the next release that should be addressed, combining the ideas I and @RichyHBM came up with we currently have 3 methods of doing this:

1. Disable all broadcast receivers by default and add a settings option to manually enable only the once actually used by the user.
2. Add a passphrase the broadcast has to contain for the backup to be triggered.
3. Add a settings item to select apps installed on the phone, only selected apps are allowed to trigger the backups.

I vote for definitely implementing the 1st option as a kind of first layer and one of the other two for additional protection.

**Comment by RichyHBM** [comment]
I agree with, and we should do both of, 1 & 2. I'm not sure if or how 3 would be achieved..

I'm hoping I can get some time over the weekend to get a PR ready with a new preferences dialog that will solve 1 and 2
